src/spartan_router.erl

```diff
@@ -13,30 +13,24 @@
 -spec(upstreams_from_questions(dns:questions()) -> ordsets:ordset(upstream())).
 upstreams_from_questions([#dns_query{name=Name}]) ->
     Labels = spartan_app:parse_upstream_name(Name),
-    lists:map(fun normalize_ip/1, find_upstream(Name, Labels));
```

Comment:
Personally, I would still keep the normalize_ip around even if we feel it is not required. You can check the tuple as part of normalize_ip.